src/main/golo/assertions.golo

```diff
@@ -0,0 +1,293 @@
+# ............................................................................................... #
+#
+# Copyright (c) 2012-2016 Institut National des Sciences Appliquées de Lyon (INSA-Lyon)
+#
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# ............................................................................................... #
+
+----
+This module provides helper functions to deal assertions.
+----
+module gololang.Assertions
+
+import gololang.Errors
+import gololang.AnsiCodes
+
+----
+List all files of a directory and sub-directories
+
+- *param* `path` the path to parse (String)
+- *param* `extension` extension of files (ie: txt or *) (String)
+- *param* `listOfFiles` initial list (generally empty) to be populated (list[])
+- *param* `root` root path to parse (initially root == path) (String)
+- *return* the list of files (list[])
+----
+local function filesDiscover = |path, extension, listOfFiles, root| {
+  let start_root = java.io.File(path)
+  let files = start_root: listFiles(): asList()
+
+  files: each(|file|{
+    if file: isDirectory() is true {
+      filesDiscover(file: getAbsolutePath(), extension, listOfFiles, root)
+    } else {
+      if file: getAbsoluteFile(): getName(): endsWith("." + extension) is true {
+        var fileName = file: getAbsolutePath(): toString(): split(root): get(1)
+        listOfFiles: add(fileName)
+      }
+    }
+  })
+  return listOfFiles
+}
+# this is the list of golo files in a project
+let goloFiles = list[]
+
+struct testsReport = {
+  passed, failed
+}
+# instance of testsReport to keep results of assertions
+let report = testsReport(0,0)
+----
+Display results of assertions
+----
+function displayTestsReport = {
+  println("tests: passed:"+report: passed()+" failed:"+report: failed())
+}
+----
+get the results of assertions
+
+- *return* results of assertions (struct testsReport)
+----
+function getTestsReport = -> report
+
+# We need to know if we can use ansi codes
+let not_windows = likelySupported()
```

Comment:
Maybe a wrapping function would be convenient instead of this global state and tests everywhere. For instance:
```golo
local function ifSupported = |f| {
    if likelySupported() {
        return f()
    }
}
```

used as:
```
...
ifSupported(-> fg_blue())
...
```

On the other hand, this will create a lot of closures. If it's a problem, you can create real functions for these operations (more cumbersome, but probably better):

```
local function fg_blue() = {
    if likelySupported() {
        gololang.AnsiCodes.fg_blue()
    }
}
# and so on...
```